service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java

```diff
@@ -0,0 +1,374 @@
+        assertThatThrownBy(() -> service.findHunterById(null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findMushroomHunterWithNonExistingId() throws Exception {
+        assertThat(service.findHunterById(-1L)).isNull();
+    }
+
+    @Test
+    public void findAllMushroomHunters() throws Exception {
+        mushroomHunterDao.create(newHunter);
+        assertThat(service.findAllHunters()).containsExactlyInAnyOrder(persistedHunter,persistedHunter2, newHunter);
+    }
+
+    @Test
+    public void findMushroomHunterByNickname() throws Exception {
+        String existingNickname = persistedHunter.getUserNickname();
+
+        assertThat(service.findHunterByNickname(existingNickname)).isNotNull();
+        assertThat(service.findHunterByNickname(existingNickname)).isEqualTo(persistedHunter);
+    }
+
+    @Test
+    public void findByMushroomHunterNullNickname() throws Exception {
+        assertThatThrownBy(() -> service.findHunterByNickname(null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findByMushroomHunterNonExistingNickname() throws Exception {
+        String nonPersistedNickname = "nickname";
+        assertThat(service.findHunterByNickname(nonPersistedNickname)).isNull();
+    }
+
+    @Test
+    public void createMushroomHunter() throws Exception {
+        service.registerHunter(newHunter, newHunter.getPasswordHash());
+        assertThat(mushroomHunterDao.findById(newHunter.getId())).isEqualTo(newHunter);
+    }
+
+    @Test
+    public void createNullMushroomHunter() throws Exception {
+        assertThatThrownBy(() -> mushroomHunterDao.create(null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void createNullFirstNameMushroomHunter() throws Exception {
+        newHunter.setFirstName(null);
+        assertThatThrownBy(() -> mushroomHunterDao.create(newHunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void createNullLSurnameMushroomHunter() throws Exception {
+        newHunter.setSurname(null);
+        assertThatThrownBy(() -> mushroomHunterDao.create(newHunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void createNullNicknameMushroomHunter() throws Exception {
+        newHunter.setUserNickname(null);
+        assertThatThrownBy(() -> mushroomHunterDao.create(newHunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void createDuplicateMushroomHunter() throws Exception {
+        assertThatThrownBy(() -> mushroomHunterDao.create(persistedHunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void deleteMushroomHunter() throws Exception {
+        assertThat(mushroomHunterDao.findById(persistedHunter.getId())).isNotNull();
+        service.deleteHunter(persistedHunter);
+        assertThat(mushroomHunterDao.findById(persistedHunter.getId())).isNull();
```

Comment:
dependent test - delete + find; use the map

Reply:
resolved